From f19df5e7c81041a4f80e03c8e2310231a1b79d35 Mon Sep 17 00:00:00 2001
From: Alexandru Croitor <alexandru.croitor@qt.io>
Date: Thu, 14 Nov 2024 17:14:24 +0100
Subject: [PATCH] CMake: Create separate SBOM documents for QtWebEngine and
 QtPdf

Because QtWebEngine and QtPdf can be installed as separate components
in the online installer, as well as because they are shipped as
separate packages in Boot2Qt, create two separate SBOM documents for
them.

Pick-to: 6.8
Fixes: QTBUG-128893
Task-number: QTBUG-122899
Task-number: QTBUG-129901
Change-Id: Icdcc8766e3338304ae7d85fa777ca74a59c14b06
Reviewed-by: Alexey Edelev <alexey.edelev@qt.io>
(cherry picked from commit a29566090106af219f3b19b342ef5005e130b336)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 CMakeLists.txt     | 3 +++
 src/CMakeLists.txt | 9 +++++++++
 2 files changed, 12 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c600f450154..3036e5452f8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,4 +48,7 @@ if(MATRIX_BUILD AND NOT MATRIX_SUBBUILD AND NOT QT_SUPERBUILD)
     return()
 endif()
 
+# Don't auto create the SBOM projects, they will be manually created
+# for each sub-project.
+set(QT_SKIP_SBOM_AUTO_PROJECT TRUE)
 qt_build_repo()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index cfe2721c995..2104424ca60 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -22,6 +22,8 @@ endif()
 #  MODULES
 ##
 
+# Use implicit PROJECT_NAME.
+qt_internal_sbom_begin_qt_repo_project()
 if(QT_FEATURE_qtwebengine_core_build)
     add_subdirectory(core)
     add_subdirectory(process)
@@ -36,7 +38,9 @@ endif()
 if(QT_FEATURE_qtwebengine_quick_build)
     add_subdirectory(webenginequick)
 endif()
+qt_internal_sbom_end_qt_repo_project()
 
+qt_internal_sbom_begin_qt_repo_project(SBOM_PROJECT_NAME QtPdf)
 if(QT_FEATURE_qtpdf_build)
     add_subdirectory(pdf)
 endif()
@@ -61,6 +65,11 @@ if(QT_FEATURE_qtpdf_build)
     endif()
 endif()
 
+# No SBOM information will be generated for any targets created past this point in the file
+# or any add_subdirectory calls. For the external project GN, we'll have to come up with some
+# additional API.
+qt_internal_sbom_end_qt_repo_project()
+
 set(installDir ${PROJECT_BINARY_DIR}/install)
 
 ##
